new-frontiers-14/frontier-station-14
# Chem Dispenser: Upgradable number of slots

## About the PR



Makes the chem dispenser, soda dispenser and booze dispensers upgradable.  The capacitor in the recipe was swapped out for a matter bin, and each tier of matter bin upgrades the total storage capacity of the machine by 5 slots (T1: 25, T2: 30, T3: 35, T4: 40).

## Why / Balance


Nice to have, gives more upgrades.

## How to test


1. Spawn a full chem dispenser.
2. Confirm it has all expected chemicals.  Insert jugs until the chem dispenser is full.
3. Upgrade the chem dispenser.  Ensure no jugs were destroyed.
4. Add jugs to fill - the number should match the description (20 + 5 * tier)

## Media


The following picture shows a full chemical dispenser with tier 4 upgrades.  The machine is storing 40 jugs.
![image](https://github.com/user-attachments/assets/80c783fa-8025-4350-a505-eb487fe33bef)

## Requirements



## Breaking changes


**Changelog**


:cl:
- tweak: Chem, booze and soda dispenser construction now require one matter bin and no capacitors.
- add: Reagent dispenser can be upgraded to increase the number of slots in the dispenser.
